pages/give-devices.js

```diff
@@ -21,77 +20,76 @@ const COUNTRIES = [
   { value: 'PH', label: 'Philippines' },
 ];
 
+const formDataInitialState = {
+  name: '',
+  email: '',
+  country: COUNTRIES[0].value,
+  streetAddress: '',
+  streetAddress2: '',
+  city: '',
+  state: '',
+  postalCode: '',
+  phone: '',
+  originalPurchasePrice: '',
+  description: '',
+};
+
+const isEmptyValidator = (value) => value.trim().length !== 0;
+
+const validations = {
+  name: {
+    error: 'Please add your name.',
+    validator: isEmptyValidator,
+  },
+  email: {
+    error: 'Please add your email address.',
+    validator: isEmptyValidator,
+  },
+  streetAddress: {
+    error: 'Please add your street address.',
+    validator: isEmptyValidator,
+  },
+  city: {
+    error: 'Please add your city.',
+    validator: isEmptyValidator,
+  },
+  state: {
+    error: 'Please add your state or province.',
+    validator: isEmptyValidator,
+  },
+  postalCode: {
+    error: 'Please add your postal code.',
+    validator: isEmptyValidator,
+  },
+};
```

Comment:
I like how you combined the validations into this object. It's really nice and readable.